ennead/models/file.py

```diff
@@ -0,0 +1,48 @@
+"""Module for Ennead file model"""
+
+import os
+import uuid
+
+from peewee import CharField, FixedCharField
+
+from ennead.models.base import BaseModel
+
+
+class FileCreationError(Exception):
+    """Error occured while creating file"""
+
+
+class File(BaseModel):
+    """One file, uploaded to server
+
+    Attributes:
+        path: real path to file on server
+        uuid: file UUID as string
+        name: filename as specified on upload
+    """
+
+    path: str = CharField()
+    uuid: str = FixedCharField(36)
+    name: str = CharField()
+
+    @classmethod
+    def from_data(cls, directory: str, name: str, data: bytes) -> 'File':
+        """Save file to directory and create DB entry for it"""
+
+        file_uuid = str(uuid.uuid4())
+        dir_path = os.path.join(os.path.realpath(directory), file_uuid)
+        os.mkdir(dir_path)
+        full_path = os.path.normpath(os.path.join(dir_path, name))
+        if not full_path.startswith(dir_path):
```

Comment:
Формально немного лучше, чтобы нельзя было записать файл в другую папку с тем же префиксом
```suggestion
        if not full_path.startswith(dir_path + os.sep):
```
Но атаки я придумать не смог: для этого, кажется, придется угадать uuid.